distribution/pom.xml

```diff
@@ -83,6 +74,7 @@
 				<executions>
 					<execution>
 						<id>attach-javadoc</id>
+						<phase>install</phase>
```

Comment:
The reasoning I thought for doing it would be to enable mvn verify to run without having these built, by pushing them into the next phase in the pipeline, install. The ordering of plugins within a phase is only done by the order in the pom.xml files that are used, so it may not be what is expected in general from Maven.

I am not quite sure what you mean by forcing the artifact to be build or not.

Caching the .m2 directory is quite difficult in general for maven, as it doesn't distinguish the paths for locally installed packages from remote repository fetched packages. It only distinguishes them inside of the metadata files that it keeps.

It would be useful to restrict this pull request to the junit categories change, and split out the travis caching changes into another pull request, as the categories change looks fine on its own but there are some ramifications to changing the plugin phases that aren't always clear up front.